packages/backend-api/src/api/services/updateNotifications.ts

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as express from 'express';
+import {isEqual, pick} from 'lodash';
+import * as WebSocket from 'ws';
+
+import {Article, Category, Comment, User} from '@conversationai/moderator-backend-core';
+import {IArticleInstance, ICategoryInstance, IUserInstance} from '@conversationai/moderator-backend-core';
+import {logger, registerInterest} from '@conversationai/moderator-backend-core';
+
+const userFields = ['id', 'name', 'email', 'avatarURL', 'group', 'isActive'];
+
+const commonFields = ['id', 'updatedAt', 'count', 'unprocessedCount', 'unmoderatedCount', 'moderatedCount',
+  'approvedCount', 'highlightedCount', 'rejectedCount', 'deferredCount', 'flaggedCount',
+  'batchedCount', 'recommendedCount', 'assignedModerators', ];
+const categoryFields = [...commonFields, 'label'];
+const articleFields = [...commonFields, 'title', 'url', 'categoryId'];
+
+interface IGlobalSummary {
+  deferred: number;
+  users: any;
+  categories: any;
+  articles: any;
+}
+
+interface IUserSummary {
+  assignments: number;
+}
+
+interface IMessage {
+  type: 'global' | 'user';
+  data: IGlobalSummary | IUserSummary;
+}
+
+async function getGlobalSummary() {
+  const deferred = await Comment.findAndCountAll({where: { isDeferred: true }, limit: 0});
+
+  const users = await User.findAll({where: {group: ['admin', 'general']}});
+  const userdata = users.map((u: IUserInstance) => {
+    const o = u.toJSON();
+    return pick(o, userFields);
+  });
+
+  const categories = await Category.findAll({
+    where: {isActive: true},
+    include: [{ model: User, as: 'assignedModerators', attributes: ['id']}],
+  });
+  const categoryIds: Array<number> = [];
+  const categorydata = categories.map((c: ICategoryInstance) => {
+    categoryIds.push(c.id);
+    const o = c.toJSON();
+    return pick(o, categoryFields);
+  });
+
+  const articles = await Article.findAll({
+    where: {$or: [{categoryId: null}, {categoryId: categoryIds}]},
+    include: [{ model: User, as: 'assignedModerators', attributes: ['id']}],
+  });
+  const articledata = articles.map((a: IArticleInstance) => {
+    const o = a.toJSON();
```

Comment:
For cases like this without any reuse of the variable, I recommend only pulling out a variable if you want to give it a meaningful name for the code-reader to further the reader's understanding. In this case we call it just 'o', so I don't think it's that useful. I'd inline this and it'll be more readable. 

More generally, it's worth asking yourself if pick should take the original object instead of a JSON version of it. .. actually why are we converting an object to JSON? pick works on an object... is this supposed to be o.fromJSON?

Reply:
This comment obviously applied to above also.

Reply:
the toJSON method on ORM objects is probably misnamed.  We use it to "realise" the object.  The ORM object can have deferred getters. The result of toJSON is just a plain dictionary.

(I copied this method from the jsonapi code, and it took me a while to work out what it did:-P)

It might be the case that pick does the right thing...

Reply:
Right, so then it's just misnamed I think: "asDict" would be better. add a TODO ? 
I looked at pick, and I don't think it does the right thing. :)